### PR TITLE
Add parser for spec "/etc/candlepin/broker.xml"

### PR DESCRIPTION
**Pull request by huali027**
Signed-off-by: Huanhuan Li <huali@redhat.com>

**Please do not include links to any websites that are not publicly accessible. You may include non-link reference numbers to help you and your team identify non-public references.**

### All Pull Requests:

* [x] Have you followed the guidelines in our Contributing document?

### CODE REVIEW

**Comment by huali027** [comment]
@xiangce, I found if we want to provide a common interface to allow adding more filters later, it is also complicated, because we have to add namespace before all tags or attributes. So at last I just remove the namespace and then remove the sensitive data before saving it. In this way, we won't be bothered by the namespace. Please help to review if it's okay to do it in this way. For the spec, the namespace is useless.

**Comment by bfahr** [comment]
@xiangce @huali027 this PR needs to wait until #3088 is merged, and it will then require changes to the datasource.  I have not yet reviewed this PR will do so after those changes have been made.

**Comment by bfahr** [comment]
I've closed #3088, but added #3110 as a pattern to follow for custom datasources.  Please modify this PR to follow that pattern.

**Comment by huali027** [comment]
@bfahr, sure, I'll update it.

**Comment by huali027** [comment]
@xiangce please help to review

File: insights/specs/default.py
Comment (huali027) [reply insights/specs/default.py:234]: @xiangce, sure, I'll update the MR, thanks.

File: insights/tests/datasources/test_candlepin_broker.py
Comment (bfahr) [inline insights/tests/datasources/test_candlepin_broker.py:144]: @huali027 this needs to be changed.  Its also wrong in cloud_init.  This may actually cause some side effects to the base `simple_file` object.  You can either create a new simple_file object and then set the content, or just create a new class that only has a content attribute.  You could also use a mock class.

File: insights/tests/datasources/test_candlepin_broker.py
Comment (bfahr) [inline insights/tests/datasources/test_candlepin_broker.py:155]: Same comment as above.

File: insights/tests/datasources/test_candlepin_broker.py
Comment (bfahr) [inline insights/tests/datasources/test_candlepin_broker.py:163]: Same as above.

**Review by bfahr — CHANGES_REQUESTED** [review]

**Review by bfahr — APPROVED** [review]
@huali027 you need to rebase this PR to remove the conflicts, but the PR is otherwise approved.

**Pull request merged**
